pkg/api/types.go

```diff
@@ -153,6 +153,15 @@ const (
 	TerminationMessagePathDefault string = "/dev/termination-log"
 )
 
+// There is no namespace required for these kinds specified here.
+// If a kind is not enumerated here, it is assumed to have a namespace scope.
+var NoneNamespacedKinds = []string{
```

Comment:
I seem to recall another one being both, but I can't remember what it was
now.

On Mon, Oct 26, 2015 at 9:29 PM, Tim Hockin <notifications@github.com>
wrote:

> In pkg/api/types.go
> <https://github.com/kubernetes/kubernetes/pull/16146#discussion_r43083528>
> :
>
> > @@ -153,6 +153,15 @@ const (
> >  	TerminationMessagePathDefault string = "/dev/termination-log"
> >  )
> >
> > +// There is no namespace required for these kinds specified here.
> > +// If a kind is not enumerated here, it is assumed to have a namespace scope.
> > +var NoneNamespacedKinds = []string{
>
> Do we expect other kinds to be both? I can see use cases for things like
> config, but I can also see "just stick it in kube-system" + cross-namespace
> object refs..
> … <#150a78edd983623a_>
> On Mon, Oct 26, 2015 at 9:06 PM, Daniel Smith <notifications@github.com>
> wrote: In pkg/api/types.go <#16146 (comment)
> <https://github.com/kubernetes/kubernetes/pull/16146#discussion_r43082631>>
> : > @@ -153,6 +153,15 @@ const ( > TerminationMessagePathDefault string =
> "/dev/termination-log" > ) > > +// There is no namespace required for these
> kinds specified here. > +// If a kind is not enumerated here, it is assumed
> to have a namespace scope. > +var NoneNamespacedKinds = []string{ There
> needs to be a single source of truth for this, but I'm not sold on making a
> list here. Also it's not binary, some resources (events) are both. — Reply
> to this email directly or view it on GitHub <
> https://github.com/kubernetes/kubernetes/pull/16146/files#r43082631>.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/kubernetes/kubernetes/pull/16146/files#r43083528>.
>